diff --git a/templates/components/VoyageForm.html.twig b/templates/components/VoyageForm.html.twig
index 4c2f90d..c76b34b 100644
--- a/templates/components/VoyageForm.html.twig
+++ b/templates/components/VoyageForm.html.twig
@@ -1,11 +1,13 @@
-{{ form_start(form) }}
-    {{ form_widget(form) }}
+<div {{ attributes }}>
+    {{ form_start(form) }}
+        {{ form_widget(form) }}
 
-    <twig:Button
-        formnovalidate
-        variant="success"
-        class="hover:animate-wiggle"
-    >
-        {{ button_label|default('Save') }}
-    </twig:Button>
-{{ form_end(form) }}
+        <twig:Button
+            formnovalidate
+            variant="success"
+            class="hover:animate-wiggle"
+        >
+            {{ button_label|default('Save') }}
+        </twig:Button>
+    {{ form_end(form) }}
+</div>
