diff --git a/src/Twig/Components/VoyageForm.php b/src/Twig/Components/VoyageForm.php
index 11f27bfb..9d0c1f52 100644
--- a/src/Twig/Components/VoyageForm.php
+++ b/src/Twig/Components/VoyageForm.php
@@ -8,6 +8,7 @@ use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\Form\FormInterface;
 use Symfony\Component\Form\FormView;
 use Symfony\UX\LiveComponent\Attribute\AsLiveComponent;
+use Symfony\UX\LiveComponent\Attribute\LiveProp;
 use Symfony\UX\LiveComponent\ComponentWithFormTrait;
 use Symfony\UX\LiveComponent\DefaultActionTrait;
 
@@ -17,9 +18,12 @@ class VoyageForm extends AbstractController
     use DefaultActionTrait;
     use ComponentWithFormTrait;
 
+    #[LiveProp]
+    public ?Voyage $initialFormData = null;
+
     protected function instantiateForm(): FormInterface
     {
-        $voyage = $voyage ?? new Voyage();
+        $voyage = $this->initialFormData ?? new Voyage();
 
         return $this->createForm(VoyageType::class, $voyage, [
             'action' => $voyage->getId() ? $this->generateUrl('app_voyage_edit', ['id' => $voyage->getId()]) : $this->generateUrl('app_voyage_new'),
diff --git a/templates/voyage/_form.html.twig b/templates/voyage/_form.html.twig
index 170cf3e2..50e790d1 100644
--- a/templates/voyage/_form.html.twig
+++ b/templates/voyage/_form.html.twig
@@ -1 +1 @@
-<twig:VoyageForm :form="form" />
+<twig:VoyageForm :form="form" :initialFormData="voyage" />
