diff --git a/src/Twig/Components/VoyageForm.php b/src/Twig/Components/VoyageForm.php
index f2c113f..11f27bf 100644
--- a/src/Twig/Components/VoyageForm.php
+++ b/src/Twig/Components/VoyageForm.php
@@ -2,11 +2,27 @@
 
 namespace App\Twig\Components;
 
+use App\Entity\Voyage;
+use App\Form\VoyageType;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\Form\FormInterface;
 use Symfony\Component\Form\FormView;
-use Symfony\UX\TwigComponent\Attribute\AsTwigComponent;
+use Symfony\UX\LiveComponent\Attribute\AsLiveComponent;
+use Symfony\UX\LiveComponent\ComponentWithFormTrait;
+use Symfony\UX\LiveComponent\DefaultActionTrait;
 
-#[AsTwigComponent]
-class VoyageForm
+#[AsLiveComponent]
+class VoyageForm extends AbstractController
 {
-    public FormView $form;
+    use DefaultActionTrait;
+    use ComponentWithFormTrait;
+
+    protected function instantiateForm(): FormInterface
+    {
+        $voyage = $voyage ?? new Voyage();
+
+        return $this->createForm(VoyageType::class, $voyage, [
+            'action' => $voyage->getId() ? $this->generateUrl('app_voyage_edit', ['id' => $voyage->getId()]) : $this->generateUrl('app_voyage_new'),
+        ]);
+    }
 }
