diff --git a/src/Controller/MainController.php b/src/Controller/MainController.php
index 6cafa115..84cc01ce 100644
--- a/src/Controller/MainController.php
+++ b/src/Controller/MainController.php
@@ -27,6 +27,7 @@ class MainController extends AbstractController
     {
         $validSorts = ['purpose', 'leaveAt'];
         $sort = in_array($sort, $validSorts) ? $sort : 'leaveAt';
+        sleep(1);
         $pager = Pagerfanta::createForCurrentPageWithMaxPerPage(
             new QueryAdapter($voyageRepository->findBySearchQueryBuilder($query, $searchPlanets, $sort, $sortDirection)),
             $page,
diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
index 2efffe3e..0a2822ff 100644
--- a/templates/main/homepage.html.twig
+++ b/templates/main/homepage.html.twig
@@ -54,7 +54,7 @@
                 >
                 <div class="whitespace-nowrap m-2 mr-4">{{ voyages|length }} results</div>
             </form>
-            <turbo-frame id="voyage-list">
+            <turbo-frame id="voyage-list" class="aria-busy:opacity-50 aria-busy:blur-sm transition-all">
                 <div class="bg-gray-800 p-4 rounded">
                     <table class="w-full text-white">
                         <thead>
