diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
index 08487988..e5ba1eff 100644
--- a/templates/main/homepage.html.twig
+++ b/templates/main/homepage.html.twig
@@ -82,11 +82,14 @@
             </div>
             <div class="flex items-center mt-6 space-x-4">
                 {% if voyages.hasPreviousPage %}
-                    <a href="#" class="block py-2 px-4 bg-gray-700 text-white rounded hover:bg-gray-600">Previous</a>
+                    <a href="{{ pagerfanta_page_url(voyages, voyages.previousPage) }}" class="block py-2 px-4 bg-gray-700 text-white rounded hover:bg-gray-600">Previous</a>
                 {% endif %}
                 {% if voyages.hasNextPage %}
-                    <a href="#" class="block py-2 px-4 bg-gray-700 text-white rounded hover:bg-gray-600">Next</a>
+                    <a href="{{ pagerfanta_page_url(voyages, voyages.nextPage) }}" class="block py-2 px-4 bg-gray-700 text-white rounded hover:bg-gray-600">Next</a>
                 {% endif %}
+                <div class="ml-4">
+                    Page {{ voyages.currentPage }}/{{ voyages.nbPages }}
+                </div>
             </div>
         </section>
     </div>
