diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
index ae2a7d85..5ad212d1 100644
--- a/templates/main/homepage.html.twig
+++ b/templates/main/homepage.html.twig
@@ -16,6 +16,7 @@
                 method="GET"
                 action="{{ path('app_homepage') }}"
                 class="mb-6 flex justify-between"
+                data-controller="autosubmit"
             >
                 <input
                     type="search"
@@ -24,6 +25,7 @@
                     aria-label="Search voyages"
                     placeholder="Search voyages"
                     class="w-1/3 px-4 py-2 rounded bg-gray-800 text-white placeholder-gray-400"
+                    data-action="autosubmit#debouncedSubmit"
                 >
                 <div class="whitespace-nowrap m-2 mr-4">{{ voyages|length }} results</div>
             </form>
