diff --git a/src/Controller/MainController.php b/src/Controller/MainController.php
index 1ed24ef..f0f239d 100644
--- a/src/Controller/MainController.php
+++ b/src/Controller/MainController.php
@@ -27,6 +27,7 @@ class MainController extends AbstractController
     {
         $validSorts = ['purpose', 'leaveAt'];
         $sort = in_array($sort, $validSorts) ? $sort : 'leaveAt';
+        sleep(1);
         $pager = Pagerfanta::createForCurrentPageWithMaxPerPage(
             new QueryAdapter($voyageRepository->findBySearchQueryBuilder($query, $searchPlanets, $sort, $sortDirection)),
             $page,
diff --git a/tailwind.config.js b/tailwind.config.js
index cfbf534..e4bcfca 100644
--- a/tailwind.config.js
+++ b/tailwind.config.js
@@ -12,7 +12,7 @@ module.exports = {
   },
   plugins: [
     plugin(function({ addVariant }) {
-      addVariant('turbo-frame', 'turbo-frame[src] &')
+      addVariant('turbo-frame', 'turbo-frame[src] &');
     }),
   ],
 }
diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
index 8bf9929..ac3311b 100644
--- a/templates/main/homepage.html.twig
+++ b/templates/main/homepage.html.twig
@@ -54,7 +54,7 @@
                 >
                 <div class="whitespace-nowrap m-2 mr-4">{{ voyages|length }} results</div>
             </form>
-            <turbo-frame id="voyage-list">
+            <turbo-frame id="voyage-list" class="aria-busy:opacity-50 aria-busy:blur-sm">
                 <div class="bg-gray-800 p-4 rounded">
                     <table class="w-full text-white">
                         <thead>
