diff --git a/src/Controller/MainController.php b/src/Controller/MainController.php
index f7f44da..ecef2f4 100644
--- a/src/Controller/MainController.php
+++ b/src/Controller/MainController.php
@@ -21,7 +21,6 @@ class MainController extends AbstractController
         $sortDirection = $request->query->get('sortDirection', 'ASC');
         $validSorts = ['purpose', 'leaveAt'];
         $sort = in_array($sort, $validSorts) ? $sort : 'leaveAt';
-        sleep(1);
 
         $pager = Pagerfanta::createForCurrentPageWithMaxPerPage(
             new QueryAdapter($voyageRepository->findBySearchQueryBuilder($query, $sort, $sortDirection)),
