diff --git a/src/Controller/MainController.php b/src/Controller/MainController.php
index 70b3fd0..8fda034 100644
--- a/src/Controller/MainController.php
+++ b/src/Controller/MainController.php
@@ -4,6 +4,8 @@ namespace App\Controller;
 
 use App\Repository\PlanetRepository;
 use App\Repository\VoyageRepository;
+use Pagerfanta\Doctrine\ORM\QueryAdapter;
+use Pagerfanta\Pagerfanta;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
@@ -16,14 +18,19 @@ class MainController extends AbstractController
     public function homepage(
         VoyageRepository $voyageRepository,
         PlanetRepository $planetRepository,
+        #[MapQueryParameter('page')] int $page = 1,
         #[MapQueryParameter('query')] string $query = null,
         #[MapQueryParameter('planets', \FILTER_VALIDATE_INT)] array $searchPlanets = [],
     ): Response
     {
-        $voyages = $voyageRepository->findBySearch($query, $searchPlanets);
+        $pager = Pagerfanta::createForCurrentPageWithMaxPerPage(
+            new QueryAdapter($voyageRepository->findBySearchQueryBuilder($query, $searchPlanets)),
+            $page,
+            10
+        );
 
         return $this->render('main/homepage.html.twig', [
-            'voyages' => $voyages,
+            'voyages' => $pager,
             'planets' => $planetRepository->findAll(),
             'searchPlanets' => $searchPlanets,
         ]);
