diff --git a/src/Controller/MainController.php b/src/Controller/MainController.php
index f0f239d..1ed24ef 100644
--- a/src/Controller/MainController.php
+++ b/src/Controller/MainController.php
@@ -27,7 +27,6 @@ class MainController extends AbstractController
     {
         $validSorts = ['purpose', 'leaveAt'];
         $sort = in_array($sort, $validSorts) ? $sort : 'leaveAt';
-        sleep(1);
         $pager = Pagerfanta::createForCurrentPageWithMaxPerPage(
             new QueryAdapter($voyageRepository->findBySearchQueryBuilder($query, $searchPlanets, $sort, $sortDirection)),
             $page,
