diff --git a/assets/controllers/modal_controller.js b/assets/controllers/modal_controller.js
index 89e2f63..084b4ea 100644
--- a/assets/controllers/modal_controller.js
+++ b/assets/controllers/modal_controller.js
@@ -12,4 +12,10 @@ export default class extends Controller {
         this.dialogTarget.close();
         document.body.classList.remove('overflow-hidden', 'blur-sm');
     }
+
+    clickOutside(event) {
+        if (event.target === this.dialogTarget) {
+            this.dialogTarget.close();
+        }
+    }
 }
diff --git a/templates/voyage/index.html.twig b/templates/voyage/index.html.twig
index 9600cd3..e447091 100644
--- a/templates/voyage/index.html.twig
+++ b/templates/voyage/index.html.twig
@@ -6,6 +6,7 @@
 <div class="m-4 p-4 bg-gray-800 rounded-lg">
     <div
         {{ stimulus_controller('modal') }}
+        {{ stimulus_action('modal', 'clickOutside', 'click') }}
         class="flex justify-between"
     >
         <h1 class="text-xl font-semibold text-white mb-4">Voyages</h1>
