diff --git a/assets/controllers/datepicker_controller.js b/assets/controllers/datepicker_controller.js
new file mode 100644
index 00000000..a5b557a3
--- /dev/null
+++ b/assets/controllers/datepicker_controller.js
@@ -0,0 +1,24 @@
+import { Controller } from '@hotwired/stimulus';
+import { Datepicker } from 'flowbite-datepicker';
+
+/* stimulusFetch: 'lazy' */
+export default class extends Controller {
+    datepicker;
+
+    connect() {
+        this.element.type = 'text';
+        this.datepicker = new Datepicker(this.element, {
+            format: 'yyyy-mm-dd',
+            autohide: true,
+        });
+    }
+
+    disconnect() {
+        if (this.datepicker) {
+            this.datepicker.destroy();
+        }
+
+        this.element.type = 'date';
+    }
+}
+
diff --git a/src/Form/VoyageType.php b/src/Form/VoyageType.php
index 4f464c91..b57f3976 100644
--- a/src/Form/VoyageType.php
+++ b/src/Form/VoyageType.php
@@ -21,6 +21,9 @@ class VoyageType extends AbstractType
             ->add('purpose')
             ->add('leaveAt', DateType::class, [
                 'widget' => 'single_text',
+                'attr' => [
+                    'data-controller' => 'datepicker',
+                ]
             ])
             ->add('planet', null, [
                 'choice_label' => 'name',
