diff --git a/src/Form/VoyageType.php b/src/Form/VoyageType.php
index 4f464c91..918cd1ed 100644
--- a/src/Form/VoyageType.php
+++ b/src/Form/VoyageType.php
@@ -21,6 +21,10 @@ class VoyageType extends AbstractType
             ->add('purpose')
             ->add('leaveAt', DateType::class, [
                 'widget' => 'single_text',
+                'attr' => [
+                    'datepicker' => true,
+                    'datepicker-format' => 'yyyy-mm-dd',
+                ]
             ])
             ->add('planet', null, [
                 'choice_label' => 'name',
