diff --git a/src/Controller/VoyageController.php b/src/Controller/VoyageController.php
index 1e0c7f0..73a274d 100644
--- a/src/Controller/VoyageController.php
+++ b/src/Controller/VoyageController.php
@@ -30,6 +30,7 @@ class VoyageController extends AbstractController
         $voyage = new Voyage();
         $form = $this->createVoyageForm($voyage);
         $form->handleRequest($request);
+        usleep(500000);
 
         if ($form->isSubmitted() && $form->isValid()) {
             $entityManager->persist($voyage);
diff --git a/templates/voyage/_form.html.twig b/templates/voyage/_form.html.twig
index 3808122..9265eb8 100644
--- a/templates/voyage/_form.html.twig
+++ b/templates/voyage/_form.html.twig
@@ -1,4 +1,4 @@
-<turbo-frame id="voyage-form">
+<turbo-frame id="voyage-form" class="aria-busy:opacity-50">
     {{ form_start(form) }}
         {{ form_widget(form) }}
 
