diff --git a/templates/voyage/_form.html.twig b/templates/voyage/_form.html.twig
index c9b53f3c..3808122f 100644
--- a/templates/voyage/_form.html.twig
+++ b/templates/voyage/_form.html.twig
@@ -1,7 +1,9 @@
-{{ form_start(form) }}
-    {{ form_widget(form) }}
+<turbo-frame id="voyage-form">
+    {{ form_start(form) }}
+        {{ form_widget(form) }}
 
-    <button formnovalidate class="px-4 py-2 border border-transparent text-sm font-medium rounded-md text-white bg-green-600 hover:bg-green-700">
-        {{ button_label|default('Save') }}
-    </button>
-{{ form_end(form) }}
+        <button formnovalidate class="px-4 py-2 border border-transparent text-sm font-medium rounded-md text-white bg-green-600 hover:bg-green-700">
+            {{ button_label|default('Save') }}
+        </button>
+    {{ form_end(form) }}
+</turbo-frame>
