diff --git a/templates/planet/show.html.twig b/templates/planet/show.html.twig
index dae7b32..58a403b 100644
--- a/templates/planet/show.html.twig
+++ b/templates/planet/show.html.twig
@@ -36,7 +36,7 @@
         <div class="mt-2">
             <a href="{{ path('app_homepage') }}">&lt;-- Back</a>
 
-            <a href="{{ path('app_planet_edit', {'id': planet.id}) }}">Edit</a>
+            <a data-turbo-frame="_top" href="{{ path('app_planet_edit', {'id': planet.id}) }}">Edit</a>
         </div>
     </turbo-frame>
 
