diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
index 2aabe35..c87fc95 100644
--- a/templates/main/homepage.html.twig
+++ b/templates/main/homepage.html.twig
@@ -6,9 +6,9 @@
     <div class="flex">
         <aside class="hidden md:block md:w-64 bg-gray-900 px-2 py-6">
             <h2 class="text-xl text-white font-semibold mb-6 px-2">Planets</h2>
-            <div>
+            <turbo-frame id="planet-info">
                 {{ include('main/_planet_list.html.twig') }}
-            </div>
+            </turbo-frame>
         </aside>
 
         <section class="flex-1 ml-10">
