diff --git a/templates/base.html.twig b/templates/base.html.twig
index 9d6fd233..99e33ef2 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -33,7 +33,7 @@
                         <twig:block name="trigger">
                             <div
                                 class="hidden md:flex pr-10 items-center space-x-2 border-2 border-gray-900 rounded-lg p-2 bg-gray-800 text-white cursor-pointer"
-                                data-action="modal#open"
+                                data-action="click->modal#open"
                             >
                                 <svg xmlns="http://www.w3.org/2000/svg" class="h-5 w-5 text-gray-500" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round"><path stroke="none" d="M0 0h24v24H0z" fill="none"/><path d="M10 10m-7 0a7 7 0 1 0 14 0a7 7 0 1 0 -14 0"/><path d="M21 21l-6 -6"/></svg>
                                 <span class="pl-2 pr-10 text-gray-500">Search Cmd+K</span>
