diff --git a/src/Twig/Components/SearchSite.php b/src/Twig/Components/SearchSite.php
index 177255a1..264aeed2 100644
--- a/src/Twig/Components/SearchSite.php
+++ b/src/Twig/Components/SearchSite.php
@@ -5,6 +5,7 @@ namespace App\Twig\Components;
 use App\Entity\Voyage;
 use App\Repository\VoyageRepository;
 use Symfony\UX\LiveComponent\Attribute\AsLiveComponent;
+use Symfony\UX\LiveComponent\Attribute\LiveProp;
 use Symfony\UX\LiveComponent\DefaultActionTrait;
 
 #[AsLiveComponent]
@@ -12,6 +13,9 @@ class SearchSite
 {
     use DefaultActionTrait;
 
+    #[LiveProp(writable: true)]
+    public string $query = '';
+
     public function __construct(private VoyageRepository $voyageRepository)
     {
     }
@@ -21,6 +25,6 @@ class SearchSite
      */
     public function voyages(): array
     {
-        return $this->voyageRepository->findBySearch(null, [], 10);
+        return $this->voyageRepository->findBySearch($this->query, [], 10);
     }
 }
diff --git a/templates/components/SearchSite.html.twig b/templates/components/SearchSite.html.twig
index e20e69cc..ea6a8b31 100644
--- a/templates/components/SearchSite.html.twig
+++ b/templates/components/SearchSite.html.twig
@@ -5,6 +5,7 @@
         </div>
         <input
             type="search"
+            data-model="query"
             aria-label="Search site"
             placeholder="Search for anything"
             class="px-4 py-2 pl-10 rounded bg-gray-800 text-white placeholder-gray-400 w-full outline-none"
