diff --git a/templates/base.html.twig b/templates/base.html.twig
index fe55413..8a9b863 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -39,6 +39,18 @@
                                 <span class="pl-2 pr-10 text-gray-500">Search Cmd+K</span>
                             </div>
                         </twig:block>
+
+                        <div class="relative">
+                            <div class="absolute inset-y-0 left-0 pl-3 flex items-center pointer-events-none">
+                                <svg xmlns="http://www.w3.org/2000/svg" class="h-5 w-5 text-gray-500" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round"><path stroke="none" d="M0 0h24v24H0z" fill="none"/><path d="M10 10m-7 0a7 7 0 1 0 14 0a7 7 0 1 0 -14 0"/><path d="M21 21l-6 -6"/></svg>
+                            </div>
+                            <input
+                                type="search"
+                                aria-label="Search site"
+                                placeholder="Search for anything"
+                                class="px-4 py-2 pl-10 rounded bg-gray-800 text-white placeholder-gray-400 w-full outline-none"
+                            />
+                        </div>
                     </twig:Modal>
                 </nav>
             </header>
