diff --git a/templates/base.html.twig b/templates/base.html.twig
index 1ccd5a6c..b197a6de 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -49,46 +49,6 @@
             </footer>
         </div>
 
-        <div
-            data-controller="modal"
-            data-action="turbo:before-cache@window->modal#close"
-        >
-            <dialog
-                class="open:flex bg-gray-800 rounded-lg shadow-xl inset-0 w-full md:w-fit md:max-w-[50%] md:min-w-[50%] animate-fade-in backdrop:bg-slate-600 backdrop:opacity-80"
-                data-modal-target="dialog"
-                data-action="close->modal#close click->modal#clickOutside"
-            >
-                <div class="flex grow p-5">
-                    <div class="grow overflow-auto p-1">
-                        <turbo-frame
-                            id="modal"
-                            data-modal-target="dynamicContent"
-                            data-action="turbo:before-fetch-request->modal#showLoading"
-                            class="aria-busy:opacity-50 transition-opacity"
-                        >
-                            {{ include('_frameSuccessStreams.html.twig', { frame: 'modal' }) }}
-                        </turbo-frame>
-                    </div>
-                </div>
-            </dialog>
-
-            <template data-modal-target="loadingTemplate">
-                <div class="bg-space-pattern bg-cover rounded-lg p-8">
-                    <div class="space-y-2">
-                        <div class="h-4 bg-gray-700 rounded w-3/4 animate-pulse"></div>
-                        <div class="h-4 bg-gray-700 rounded animate-pulse"></div>
-                        <div class="h-4 bg-gray-700 rounded animate-pulse"></div>
-                        <div class="h-4"></div>
-                        <div class="h-4 bg-gray-700 rounded animate-pulse"></div>
-                        <div class="h-4 bg-gray-700 rounded w-1/2 animate-pulse"></div>
-                        <div class="h-4 bg-gray-700 rounded w-3/4 animate-pulse"></div>
-                        <div class="h-4"></div>
-                        <div class="h-4 bg-gray-700 rounded w-1/2 animate-pulse"></div>
-                    </div>
-                </div>
-            </template>
-        </div>
-
         <div id="flash-container">
             {{ include('_flashes.html.twig') }}
         </div>
diff --git a/templates/components/Modal.html.twig b/templates/components/Modal.html.twig
new file mode 100644
index 00000000..5d439339
--- /dev/null
+++ b/templates/components/Modal.html.twig
@@ -0,0 +1,39 @@
+<div
+    data-controller="modal"
+    data-action="turbo:before-cache@window->modal#close"
+>
+    <dialog
+        class="open:flex bg-gray-800 rounded-lg shadow-xl inset-0 w-full md:w-fit md:max-w-[50%] md:min-w-[50%] animate-fade-in backdrop:bg-slate-600 backdrop:opacity-80"
+        data-modal-target="dialog"
+        data-action="close->modal#close click->modal#clickOutside"
+    >
+        <div class="flex grow p-5">
+            <div class="grow overflow-auto p-1">
+                <turbo-frame
+                    id="modal"
+                    data-modal-target="dynamicContent"
+                    data-action="turbo:before-fetch-request->modal#showLoading"
+                    class="aria-busy:opacity-50 transition-opacity"
+                >
+                    {{ include('_frameSuccessStreams.html.twig', { frame: 'modal' }) }}
+                </turbo-frame>
+            </div>
+        </div>
+    </dialog>
+
+    <template data-modal-target="loadingTemplate">
+        <div class="bg-space-pattern bg-cover rounded-lg p-8">
+            <div class="space-y-2">
+                <div class="h-4 bg-gray-700 rounded w-3/4 animate-pulse"></div>
+                <div class="h-4 bg-gray-700 rounded animate-pulse"></div>
+                <div class="h-4 bg-gray-700 rounded animate-pulse"></div>
+                <div class="h-4"></div>
+                <div class="h-4 bg-gray-700 rounded animate-pulse"></div>
+                <div class="h-4 bg-gray-700 rounded w-1/2 animate-pulse"></div>
+                <div class="h-4 bg-gray-700 rounded w-3/4 animate-pulse"></div>
+                <div class="h-4"></div>
+                <div class="h-4 bg-gray-700 rounded w-1/2 animate-pulse"></div>
+            </div>
+        </div>
+    </template>
+</div>
