diff --git a/templates/voyage/_row.html.twig b/templates/voyage/_row.html.twig
index d376ea1d..beca337f 100644
--- a/templates/voyage/_row.html.twig
+++ b/templates/voyage/_row.html.twig
@@ -1,4 +1,4 @@
-<tr class="even:bg-gray-700 odd:bg-gray-600">
+<tr class="even:bg-gray-700 odd:bg-gray-600" id="voyage-list-item-{{ voyage.id }}">
     <td class="px-6 py-4 whitespace-nowrap">{{ voyage.id }}</td>
     <td class="px-6 py-4">{{ voyage.purpose }}</td>
     <td class="px-6 py-4 whitespace-nowrap">{{ voyage.leaveAt ? voyage.leaveAt|date('Y-m-d H:i:s') : '' }}</td>
diff --git a/templates/voyage/edit.html.twig b/templates/voyage/edit.html.twig
index 4d450562..258e81ef 100644
--- a/templates/voyage/edit.html.twig
+++ b/templates/voyage/edit.html.twig
@@ -27,4 +27,7 @@
     <turbo-stream action="update" target="modal">
         <template></template>
     </turbo-stream>
+    <turbo-stream action="replace" target="voyage-list-item-{{ voyage.id }}">
+        <template>{{ include('voyage/_row.html.twig') }}</template>
+    </turbo-stream>
 {% endblock %}
