diff --git a/templates/components/Modal.html.twig b/templates/components/Modal.html.twig
index 34b7ff9f..b158b847 100644
--- a/templates/components/Modal.html.twig
+++ b/templates/components/Modal.html.twig
@@ -1,4 +1,4 @@
-{% props allowSmallWidth=false %}
+{% props allowSmallWidth=false, closeButton=false %}
 <div
     {{ attributes.defaults({
         'data-controller': 'modal',
@@ -21,6 +21,14 @@
                 {% endblock %}
             </div>
         </div>
+        {% if closeButton %}
+            <button
+                class="absolute right-4 top-3 text-white flex items-center opacity-70 transition-opacity hover:opacity-100"
+                data-action="modal#close"
+            >
+                <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round"><path stroke="none" d="M0 0h24v24H0z" fill="none"/><path d="M18 6l-12 12" /><path d="M6 6l12 12" /></svg>
+            </button>
+        {% endif %}
     </dialog>
 
     {% if block('loading_template') %}
diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
index 85230b71..f6fa876c 100644
--- a/templates/main/homepage.html.twig
+++ b/templates/main/homepage.html.twig
@@ -55,7 +55,7 @@
                         autofocus
                     >
 
-                    <twig:Modal>
+                    <twig:Modal :closeButton="true">
                         <twig:block name="trigger">
                             <twig:Button
                                 variant="link"
