diff --git a/templates/voyage/_row.html.twig b/templates/voyage/_row.html.twig
new file mode 100644
index 00000000..1ba68b3e
--- /dev/null
+++ b/templates/voyage/_row.html.twig
@@ -0,0 +1,9 @@
+<tr class="even:bg-gray-700 odd:bg-gray-600">
+    <td class="px-6 py-4 whitespace-nowrap">{{ voyage.id }}</td>
+    <td class="px-6 py-4">{{ voyage.purpose }}</td>
+    <td class="px-6 py-4 whitespace-nowrap">{{ voyage.leaveAt ? voyage.leaveAt|date('Y-m-d H:i:s') : '' }}</td>
+    <td class="px-6 py-4 whitespace-nowrap">
+        <a href="{{ path('app_voyage_show', {'id': voyage.id}) }}" class="text-blue-400 hover:text-blue-600">show</a>
+        <a href="{{ path('app_voyage_edit', {'id': voyage.id}) }}" class="ml-4 text-yellow-400 hover:text-yellow-600">edit</a>
+    </td>
+</tr>
diff --git a/templates/voyage/index.html.twig b/templates/voyage/index.html.twig
index 5e1ef65f..11b3cbc1 100644
--- a/templates/voyage/index.html.twig
+++ b/templates/voyage/index.html.twig
@@ -30,15 +30,7 @@
         </thead>
         <tbody class="divide-y divide-gray-600">
             {% for voyage in voyages %}
-                <tr class="even:bg-gray-700 odd:bg-gray-600">
-                    <td class="px-6 py-4 whitespace-nowrap">{{ voyage.id }}</td>
-                    <td class="px-6 py-4">{{ voyage.purpose }}</td>
-                    <td class="px-6 py-4 whitespace-nowrap">{{ voyage.leaveAt ? voyage.leaveAt|date('Y-m-d H:i:s') : '' }}</td>
-                    <td class="px-6 py-4 whitespace-nowrap">
-                        <a href="{{ path('app_voyage_show', {'id': voyage.id}) }}" class="text-blue-400 hover:text-blue-600">show</a>
-                        <a href="{{ path('app_voyage_edit', {'id': voyage.id}) }}" class="ml-4 text-yellow-400 hover:text-yellow-600">edit</a>
-                    </td>
-                </tr>
+                {{ include('voyage/_row.html.twig') }}
             {% else %}
                 <tr>
                     <td colspan="4" class="px-6 py-4 whitespace-nowrap text-center text-gray-400">No records found</td>
diff --git a/templates/voyage/new.html.twig b/templates/voyage/new.html.twig
index b4d39384..02e92800 100644
--- a/templates/voyage/new.html.twig
+++ b/templates/voyage/new.html.twig
@@ -25,7 +25,7 @@
 {% block stream_success %}
     <turbo-stream action="prepend" targets="#voyage-list tbody">
         <template>
-            <tr><td>{{ voyage.purpose }}</td></tr>
+            {{ include('voyage/_row.html.twig') }}
         </template>
     </turbo-stream>
 {% endblock %}
