diff --git a/src/Controller/VoyageController.php b/src/Controller/VoyageController.php
index e772dfd5..f7d3fd4b 100644
--- a/src/Controller/VoyageController.php
+++ b/src/Controller/VoyageController.php
@@ -11,7 +11,6 @@ use Symfony\Component\Form\FormInterface;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Annotation\Route;
-use Symfony\UX\Turbo\TurboBundle;
 
 #[Route('/voyage')]
 class VoyageController extends AbstractController
@@ -38,11 +37,11 @@ class VoyageController extends AbstractController
             $this->addFlash('success', 'Bon voyage!');
 
             if ($request->headers->has('turbo-frame')) {
-                $request->setRequestFormat(TurboBundle::STREAM_FORMAT);
-
-                return $this->renderBlock('voyage/new.html.twig', 'stream_success', [
+                $stream = $this->renderBlockView('voyage/new.html.twig', 'stream_success', [
                     'voyage' => $voyage
                 ]);
+
+                $this->addFlash('stream', $stream);
             }
 
             return $this->redirectToRoute('app_voyage_index', [], Response::HTTP_SEE_OTHER);
diff --git a/templates/voyage/new.html.twig b/templates/voyage/new.html.twig
index c01fd915..02e92800 100644
--- a/templates/voyage/new.html.twig
+++ b/templates/voyage/new.html.twig
@@ -28,10 +28,4 @@
             {{ include('voyage/_row.html.twig') }}
         </template>
     </turbo-stream>
-    <turbo-stream action="update" target="modal">
-        <template></template>
-    </turbo-stream>
-    <turbo-stream action="append" target="flash-container">
-        <template>{{ include('_flashes.html.twig') }}</template>
-    </turbo-stream>
 {% endblock %}
