diff --git a/templates/base.html.twig b/templates/base.html.twig
index b197a6de..295f0137 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -49,6 +49,17 @@
             </footer>
         </div>
 
+        <twig:Modal>
+            <turbo-frame
+                id="modal"
+                data-modal-target="dynamicContent"
+                data-action="turbo:before-fetch-request->modal#showLoading"
+                class="aria-busy:opacity-50 transition-opacity"
+            >
+                {{ include('_frameSuccessStreams.html.twig', { frame: 'modal' }) }}
+            </turbo-frame>
+        </twig:Modal>
+
         <div id="flash-container">
             {{ include('_flashes.html.twig') }}
         </div>
diff --git a/templates/components/Modal.html.twig b/templates/components/Modal.html.twig
index 08d29e96..0892c609 100644
--- a/templates/components/Modal.html.twig
+++ b/templates/components/Modal.html.twig
@@ -11,14 +11,7 @@
     >
         <div class="flex grow p-5">
             <div class="grow overflow-auto p-1">
-                <turbo-frame
-                    id="modal"
-                    data-modal-target="dynamicContent"
-                    data-action="turbo:before-fetch-request->modal#showLoading"
-                    class="aria-busy:opacity-50 transition-opacity"
-                >
-                    {{ include('_frameSuccessStreams.html.twig', { frame: 'modal' }) }}
-                </turbo-frame>
+                {% block content %}{% endblock %}
             </div>
         </div>
     </dialog>
