diff --git a/src/Controller/VoyageController.php b/src/Controller/VoyageController.php
index 36159d74..f7d3fd4b 100644
--- a/src/Controller/VoyageController.php
+++ b/src/Controller/VoyageController.php
@@ -92,10 +92,19 @@ class VoyageController extends AbstractController
     public function delete(Request $request, Voyage $voyage, EntityManagerInterface $entityManager): Response
     {
         if ($this->isCsrfTokenValid('delete'.$voyage->getId(), $request->request->get('_token'))) {
+            $id = $voyage->getId();
             $entityManager->remove($voyage);
             $entityManager->flush();
 
             $this->addFlash('success', 'Voyage deleted!');
+
+            if ($request->headers->has('turbo-frame')) {
+                $stream = $this->renderBlockView('voyage/delete.html.twig', 'success_stream', [
+                    'id' => $id,
+                ]);
+
+                $this->addFlash('stream', $stream);
+            }
         }
 
         return $this->redirectToRoute('app_voyage_index', [], Response::HTTP_SEE_OTHER);
diff --git a/templates/voyage/delete.html.twig b/templates/voyage/delete.html.twig
new file mode 100644
index 00000000..80635fc3
--- /dev/null
+++ b/templates/voyage/delete.html.twig
@@ -0,0 +1,3 @@
+{% block success_stream %}
+    <turbo-stream action="remove" target="voyage-list-item-{{ id }}"></turbo-stream>
+{% endblock %}
