diff --git a/templates/base.html.twig b/templates/base.html.twig
index 8914f81..7981f89 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -55,39 +55,6 @@
             {{ include('_flashes.html.twig') }}
         </div>
 
-        <div
-            {{ stimulus_controller('modal') }}
-            {{ stimulus_action('modal', 'clickOutside', 'click') }}
-        >
-            <dialog
-                class="bg-gray-800 rounded-lg shadow-xl inset-0 w-full lg:w-1/2 animate-fade-in backdrop:bg-slate-600 backdrop:opacity-80"
-                {{ stimulus_target('modal', 'dialog') }}
-                {{ stimulus_action('modal', 'close', 'close') }}
-            >
-                <div class="text-white space-y-4 p-5">
-                    <turbo-frame
-                        id="modal"
-                        {{ stimulus_target('modal', 'modalContent') }}
-                        {{ stimulus_action('modal', 'showLoading', 'turbo:before-fetch-request') }}
-                    ></turbo-frame>
-                </div>
-            </dialog>
-
-            <template {{ stimulus_target('modal', 'loadingTemplate') }}>
-                <div class="bg-space-pattern bg-cover rounded-lg p-8">
-                    <div class="space-y-2">
-                        <div class="h-4 bg-gray-700 rounded w-3/4 animate-pulse"></div>
-                        <div class="h-4 bg-gray-700 rounded animate-pulse"></div>
-                        <div class="h-4 bg-gray-700 rounded animate-pulse"></div>
-                        <div class="h-4"></div>
-                        <div class="h-4 bg-gray-700 rounded animate-pulse"></div>
-                        <div class="h-4 bg-gray-700 rounded w-1/2 animate-pulse"></div>
-                        <div class="h-4 bg-gray-700 rounded w-3/4 animate-pulse"></div>
-                        <div class="h-4"></div>
-                        <div class="h-4 bg-gray-700 rounded w-1/2 animate-pulse"></div>
-                    </div>
-                </div>
-            </template>
-        </div>
+        <twig:Modal />
     </body>
 </html>
diff --git a/templates/components/Modal.html.twig b/templates/components/Modal.html.twig
new file mode 100644
index 0000000..94460c3
--- /dev/null
+++ b/templates/components/Modal.html.twig
@@ -0,0 +1,34 @@
+<div
+    {{ stimulus_controller('modal') }}
+    {{ stimulus_action('modal', 'clickOutside', 'click') }}
+>
+    <dialog
+        class="bg-gray-800 rounded-lg shadow-xl inset-0 w-full lg:w-1/2 animate-fade-in backdrop:bg-slate-600 backdrop:opacity-80"
+        {{ stimulus_target('modal', 'dialog') }}
+        {{ stimulus_action('modal', 'close', 'close') }}
+    >
+        <div class="text-white space-y-4 p-5">
+            <turbo-frame
+                id="modal"
+                {{ stimulus_target('modal', 'modalContent') }}
+                {{ stimulus_action('modal', 'showLoading', 'turbo:before-fetch-request') }}
+            ></turbo-frame>
+        </div>
+    </dialog>
+
+    <template {{ stimulus_target('modal', 'loadingTemplate') }}>
+        <div class="bg-space-pattern bg-cover rounded-lg p-8">
+            <div class="space-y-2">
+                <div class="h-4 bg-gray-700 rounded w-3/4 animate-pulse"></div>
+                <div class="h-4 bg-gray-700 rounded animate-pulse"></div>
+                <div class="h-4 bg-gray-700 rounded animate-pulse"></div>
+                <div class="h-4"></div>
+                <div class="h-4 bg-gray-700 rounded animate-pulse"></div>
+                <div class="h-4 bg-gray-700 rounded w-1/2 animate-pulse"></div>
+                <div class="h-4 bg-gray-700 rounded w-3/4 animate-pulse"></div>
+                <div class="h-4"></div>
+                <div class="h-4 bg-gray-700 rounded w-1/2 animate-pulse"></div>
+            </div>
+        </div>
+    </template>
+</div>
