diff --git a/templates/base.html.twig b/templates/base.html.twig
index 0b96a76..9883c3d 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -68,6 +68,7 @@
                             id="modal"
                             data-modal-target="dynamicContent"
                             data-action="turbo:before-fetch-request->modal#showLoading"
+                            class="aria-busy:opacity-50 transition-opacity"
                         ></turbo-frame>
                     </div>
                 </div>
