diff --git a/templates/voyage/_row.html.twig b/templates/voyage/_row.html.twig
index beca337f..01c5f1d1 100644
--- a/templates/voyage/_row.html.twig
+++ b/templates/voyage/_row.html.twig
@@ -9,5 +9,10 @@
             class="ml-4 text-yellow-400 hover:text-yellow-600"
             data-turbo-frame="modal"
         >edit</a>
+        <a
+            href="{{ path('app_voyage_edit', {'id': voyage.id}) }}"
+            class="ml-4 text-yellow-400 hover:text-yellow-600"
+            data-turbo-frame="modal"
+        >edit</a>
     </td>
 </tr>
