diff --git a/templates/base.html.twig b/templates/base.html.twig
index cce59991..ed61d731 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -49,10 +49,6 @@
             </footer>
         </div>
 
-        <div id="flash-container">
-            {{ include('_flashes.html.twig') }}
-        </div>
-
         <div
             data-controller="modal"
             data-action="turbo:before-cache@window->modal#close"
@@ -94,5 +90,9 @@
                 </div>
             </template>
         </div>
+
+        <div id="flash-container">
+            {{ include('_flashes.html.twig') }}
+        </div>
     </body>
 </html>
