diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
index 7b4316cf..4006ae6d 100644
--- a/templates/main/homepage.html.twig
+++ b/templates/main/homepage.html.twig
@@ -53,6 +53,26 @@
                         data-action="autosubmit#debouncedSubmit"
                         autofocus
                     >
+
+                    <twig:Modal>
+                        <twig:block name="trigger">
+                            <twig:Button
+                                variant="link"
+                                type="button"
+                                data-action="modal#open"
+                            >Options</twig:Button>
+                        </twig:block>
+
+                        <h3 class="text-white text-lg font-semibold mb-2">Search Options</h3>
+                        <hr class="mb-4">
+
+                        <div class="flex justify-end">
+                            <twig:Button
+                                variant="success"
+                                data-action="modal#close"
+                            >See Results</twig:Button>
+                        </div>
+                    </twig:Modal>
                 </div>
                 {% block result_count %}
                     <div id="voyage-result-count" class="whitespace-nowrap m-2 mr-4">{{ voyages|length }} results</div>
