diff --git a/assets/controllers/modal_controller.js b/assets/controllers/modal_controller.js
index 084b4eac..62409eba 100644
--- a/assets/controllers/modal_controller.js
+++ b/assets/controllers/modal_controller.js
@@ -1,7 +1,38 @@
 import { Controller } from '@hotwired/stimulus';
 
 export default class extends Controller {
-    static targets = ['dialog'];
+    static targets = ['dialog', 'dynamicContent'];
+
+    observer = null;
+
+    connect() {
+        if (this.hasDynamicContentTarget) {
+            // when the content changes, call this.open()
+            this.observer = new MutationObserver(() => {
+                const shouldOpen = this.dynamicContentTarget.innerHTML.trim().length > 0;
+
+                if (shouldOpen && !this.dialogTarget.open) {
+                    this.open();
+                } else if (!shouldOpen && this.dialogTarget.open) {
+                    this.close();
+                }
+            });
+            this.observer.observe(this.dynamicContentTarget, {
+                childList: true,
+                characterData: true,
+                subtree: true
+            });
+        }
+    }
+
+    disconnect() {
+        if (this.observer) {
+            this.observer.disconnect();
+        }
+        if (this.dialogTarget.open) {
+            this.close();
+        }
+    }
 
     open() {
         this.dialogTarget.showModal();
diff --git a/templates/base.html.twig b/templates/base.html.twig
index dbffebd5..b0825425 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -64,7 +64,7 @@
             >
                 <div class="flex grow p-5">
                     <div class="grow overflow-auto p-1">
-                        <turbo-frame id="modal"></turbo-frame>
+                        <turbo-frame id="modal" data-modal-target="dynamicContent"></turbo-frame>
                     </div>
                 </div>
             </dialog>
