diff --git a/src/Controller/VoyageController.php b/src/Controller/VoyageController.php
index 0e9dc828..08be053d 100644
--- a/src/Controller/VoyageController.php
+++ b/src/Controller/VoyageController.php
@@ -29,7 +29,6 @@ class VoyageController extends AbstractController
         $voyage = new Voyage();
         $form = $this->createVoyageForm($voyage);
         $form->handleRequest($request);
-        sleep(2);
 
         if ($form->isSubmitted() && $form->isValid()) {
             $entityManager->persist($voyage);
