diff --git a/assets/controllers/modal_controller.js b/assets/controllers/modal_controller.js
index 62409eb..aa3a7fe 100644
--- a/assets/controllers/modal_controller.js
+++ b/assets/controllers/modal_controller.js
@@ -1,7 +1,7 @@
 import { Controller } from '@hotwired/stimulus';
 
 export default class extends Controller {
-    static targets = ['dialog', 'dynamicContent'];
+    static targets = ['dialog', 'dynamicContent', 'loadingContent'];
 
     observer = null;
 
@@ -49,4 +49,13 @@ export default class extends Controller {
             this.dialogTarget.close();
         }
     }
+
+    showLoading() {
+        // do nothing if the dialog is already open
+        if (this.dialogTarget.open) {
+            return;
+        }
+
+        this.dynamicContentTarget.innerHTML = this.loadingContentTarget.innerHTML;
+    }
 }
diff --git a/src/Controller/VoyageController.php b/src/Controller/VoyageController.php
index 08be053..0e9dc82 100644
--- a/src/Controller/VoyageController.php
+++ b/src/Controller/VoyageController.php
@@ -29,6 +29,7 @@ class VoyageController extends AbstractController
         $voyage = new Voyage();
         $form = $this->createVoyageForm($voyage);
         $form->handleRequest($request);
+        sleep(2);
 
         if ($form->isSubmitted() && $form->isValid()) {
             $entityManager->persist($voyage);
diff --git a/templates/base.html.twig b/templates/base.html.twig
index b082542..0b96a76 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -64,10 +64,30 @@
             >
                 <div class="flex grow p-5">
                     <div class="grow overflow-auto p-1">
-                        <turbo-frame id="modal" data-modal-target="dynamicContent"></turbo-frame>
+                        <turbo-frame
+                            id="modal"
+                            data-modal-target="dynamicContent"
+                            data-action="turbo:before-fetch-request->modal#showLoading"
+                        ></turbo-frame>
                     </div>
                 </div>
             </dialog>
+
+            <template data-modal-target="loadingContent">
+                <div class="bg-space-pattern bg-cover rounded-lg p-8">
+                    <div class="space-y-2">
+                        <div class="h-4 bg-gray-700 rounded w-3/4 animate-pulse"></div>
+                        <div class="h-4 bg-gray-700 rounded animate-pulse"></div>
+                        <div class="h-4 bg-gray-700 rounded animate-pulse"></div>
+                        <div class="h-4"></div>
+                        <div class="h-4 bg-gray-700 rounded animate-pulse"></div>
+                        <div class="h-4 bg-gray-700 rounded w-1/2 animate-pulse"></div>
+                        <div class="h-4 bg-gray-700 rounded w-3/4 animate-pulse"></div>
+                        <div class="h-4"></div>
+                        <div class="h-4 bg-gray-700 rounded w-1/2 animate-pulse"></div>
+                    </div>
+                </div>
+            </template>
         </div>
     </body>
 </html>
