diff --git a/templates/components/Modal.html.twig b/templates/components/Modal.html.twig
index 90e6fc44..34b7ff9f 100644
--- a/templates/components/Modal.html.twig
+++ b/templates/components/Modal.html.twig
@@ -1,3 +1,4 @@
+{% props allowSmallWidth=false %}
 <div
     {{ attributes.defaults({
         'data-controller': 'modal',
@@ -7,7 +8,7 @@
     {% block trigger %}{% endblock %}
 
     <dialog
-        class="open:flex bg-gray-800 rounded-lg shadow-xl inset-0 w-full md:w-fit md:max-w-[50%] md:min-w-[50%] animate-fade-in backdrop:bg-slate-600 backdrop:opacity-80"
+        class="open:flex bg-gray-800 rounded-lg shadow-xl inset-0 w-full md:w-fit md:max-w-[50%] {{ allowSmallWidth ? '' : 'md:min-w-[50%] ' }}animate-fade-in backdrop:bg-slate-600 backdrop:opacity-80"
         data-modal-target="dialog"
         data-action="close->modal#close"
     >
diff --git a/templates/voyage/_row.html.twig b/templates/voyage/_row.html.twig
index aede6241..2e3f3c3d 100644
--- a/templates/voyage/_row.html.twig
+++ b/templates/voyage/_row.html.twig
@@ -10,7 +10,7 @@
             data-turbo-frame="modal"
         >edit</a>
 
-        <twig:Modal>
+        <twig:Modal :allowSmallWidth="true">
             <twig:block name="trigger">
                 <button
                     class="ml-4 text-red-400 hover:text-red-600"
