diff --git a/templates/voyage/_row.html.twig b/templates/voyage/_row.html.twig
index 1ba68b3..d376ea1 100644
--- a/templates/voyage/_row.html.twig
+++ b/templates/voyage/_row.html.twig
@@ -4,6 +4,10 @@
     <td class="px-6 py-4 whitespace-nowrap">{{ voyage.leaveAt ? voyage.leaveAt|date('Y-m-d H:i:s') : '' }}</td>
     <td class="px-6 py-4 whitespace-nowrap">
         <a href="{{ path('app_voyage_show', {'id': voyage.id}) }}" class="text-blue-400 hover:text-blue-600">show</a>
-        <a href="{{ path('app_voyage_edit', {'id': voyage.id}) }}" class="ml-4 text-yellow-400 hover:text-yellow-600">edit</a>
+        <a
+            href="{{ path('app_voyage_edit', {'id': voyage.id}) }}"
+            class="ml-4 text-yellow-400 hover:text-yellow-600"
+            data-turbo-frame="modal"
+        >edit</a>
     </td>
 </tr>
diff --git a/templates/voyage/edit.html.twig b/templates/voyage/edit.html.twig
index 81e13f9..a2e22e5 100644
--- a/templates/voyage/edit.html.twig
+++ b/templates/voyage/edit.html.twig
@@ -1,4 +1,4 @@
-{% extends 'base.html.twig' %}
+{% extends 'modalBase.html.twig' %}
 
 {% block title %}Edit Voyage{% endblock %}
 
@@ -24,9 +24,5 @@
 {% endblock %}
 
 {% block success_stream %}
-    <turbo-stream action="append" target="flash-container">
-        <template>
-            {{ include('_flashes.html.twig') }}
-        </template>
-    </turbo-stream>
+    {{ include('_streamSuccess.html.twig') }}
 {% endblock %}
