diff --git a/templates/voyage/index.html.twig b/templates/voyage/index.html.twig
index a916d885..e273a3b3 100644
--- a/templates/voyage/index.html.twig
+++ b/templates/voyage/index.html.twig
@@ -51,8 +51,6 @@
         </dialog>
     </div>
 
-    {{ include('voyage/_form.html.twig') }}
-
     <table class="min-w-full bg-gray-800 text-white" id="voyage-list">
         <thead class="border-b border-gray-700">
             <tr>
