diff --git a/src/Twig/AppExtension.php b/src/Twig/AppExtension.php
index 5f4892d..aaff80c 100644
--- a/src/Twig/AppExtension.php
+++ b/src/Twig/AppExtension.php
@@ -16,6 +16,7 @@ class AppExtension extends AbstractExtension
     {
         return [
             'is_frame_request' => new TwigFunction('is_frame_request', [$this, 'isFrameRequest']),
+            'conditional_frame_id' => new TwigFunction('conditional_frame_id', [$this, 'conditionalFrameId']),
         ];
     }
 
@@ -32,4 +33,13 @@ class AppExtension extends AbstractExtension
 
         return $request->headers->get('Turbo-Frame') === $frameId;
     }
+
+    public function conditionalFrameId(string $frameId): string
+    {
+        if ($this->isFrameRequest($frameId)) {
+            return $frameId;
+        }
+
+        return sprintf('%s-page', $frameId);
+    }
 }
diff --git a/templates/voyage/new.html.twig b/templates/voyage/new.html.twig
index fd201ae..79cfe75 100644
--- a/templates/voyage/new.html.twig
+++ b/templates/voyage/new.html.twig
@@ -3,7 +3,7 @@
 {% block title %}New Voyage{% endblock %}
 
 {% block body %}
-    <turbo-frame id="{{ is_frame_request('modal') ? 'modal' : 'modal-page' }}" target="_top" class="block m-4 p-4 bg-gray-800 rounded-lg">
+    <turbo-frame id="{{ conditional_frame_id('modal') }}" target="_top" class="block m-4 p-4 bg-gray-800 rounded-lg">
         <div class="flex justify-between">
             <h1 class="text-xl font-semibold text-white mb-4">Create New Voyage</h1>
 
