diff --git a/templates/base.html.twig b/templates/base.html.twig
index 9ed74272..8428a9e0 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -61,6 +61,21 @@
             >
                 {{ include('_frameSuccessStreams.html.twig', { frame: 'modal' }) }}
             </turbo-frame>
+            <twig:block name="loading_template">
+                <div class="bg-space-pattern bg-cover rounded-lg p-8">
+                    <div class="space-y-2">
+                        <div class="h-4 bg-gray-700 rounded w-3/4 animate-pulse"></div>
+                        <div class="h-4 bg-gray-700 rounded animate-pulse"></div>
+                        <div class="h-4 bg-gray-700 rounded animate-pulse"></div>
+                        <div class="h-4"></div>
+                        <div class="h-4 bg-gray-700 rounded animate-pulse"></div>
+                        <div class="h-4 bg-gray-700 rounded w-1/2 animate-pulse"></div>
+                        <div class="h-4 bg-gray-700 rounded w-3/4 animate-pulse"></div>
+                        <div class="h-4"></div>
+                        <div class="h-4 bg-gray-700 rounded w-1/2 animate-pulse"></div>
+                    </div>
+                </div>
+            </twig:block>
         </twig:Modal>
     </body>
 </html>
diff --git a/templates/components/Modal.html.twig b/templates/components/Modal.html.twig
index 90eafe5b..f763c21d 100644
--- a/templates/components/Modal.html.twig
+++ b/templates/components/Modal.html.twig
@@ -16,19 +16,9 @@
         </div>
     </dialog>
 
-    <template data-modal-target="loadingTemplate">
-        <div class="bg-space-pattern bg-cover rounded-lg p-8">
-            <div class="space-y-2">
-                <div class="h-4 bg-gray-700 rounded w-3/4 animate-pulse"></div>
-                <div class="h-4 bg-gray-700 rounded animate-pulse"></div>
-                <div class="h-4 bg-gray-700 rounded animate-pulse"></div>
-                <div class="h-4"></div>
-                <div class="h-4 bg-gray-700 rounded animate-pulse"></div>
-                <div class="h-4 bg-gray-700 rounded w-1/2 animate-pulse"></div>
-                <div class="h-4 bg-gray-700 rounded w-3/4 animate-pulse"></div>
-                <div class="h-4"></div>
-                <div class="h-4 bg-gray-700 rounded w-1/2 animate-pulse"></div>
-            </div>
-        </div>
-    </template>
+    {% if block('loading_template') %}
+        <template data-modal-target="loadingTemplate">
+            {% block loading_template %}{% endblock %}
+        </template>
+    {% endif %}
 </div>
