diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
index 92c67bb..7b4316c 100644
--- a/templates/main/homepage.html.twig
+++ b/templates/main/homepage.html.twig
@@ -42,16 +42,18 @@
                 data-controller="autosubmit"
                 data-turbo-frame="voyage-list"
             >
-                <input
-                    type="search"
-                    name="query"
-                    value="{{ app.request.query.get('query') }}"
-                    aria-label="Search voyages"
-                    placeholder="Search voyages"
-                    class="w-1/3 px-4 py-2 rounded bg-gray-800 text-white placeholder-gray-400"
-                    data-action="autosubmit#debouncedSubmit"
-                    autofocus
-                >
+                <div class="w-1/3 flex">
+                    <input
+                        type="search"
+                        name="query"
+                        value="{{ app.request.query.get('query') }}"
+                        aria-label="Search voyages"
+                        placeholder="Search voyages"
+                        class="px-4 py-2 rounded bg-gray-800 text-white placeholder-gray-400"
+                        data-action="autosubmit#debouncedSubmit"
+                        autofocus
+                    >
+                </div>
                 {% block result_count %}
                     <div id="voyage-result-count" class="whitespace-nowrap m-2 mr-4">{{ voyages|length }} results</div>
                 {% endblock %}
