diff --git a/templates/voyage/index.html.twig b/templates/voyage/index.html.twig
index 3eb438f..4289d4e 100644
--- a/templates/voyage/index.html.twig
+++ b/templates/voyage/index.html.twig
@@ -9,13 +9,14 @@
     >
         <h1 class="text-xl font-semibold text-white mb-4">Voyages</h1>
 
-        <button
-            data-action="modal#open"
+        <a
+            href="{{ path('app_voyage_new') }}"
+            data-turbo-frame="modal"
             class="flex items-center space-x-1 bg-blue-500 hover:bg-blue-700 text-white text-sm font-bold px-4 rounded"
         >
             <span>New Voyage</span>
             <svg xmlns="http://www.w3.org/2000/svg" class="w-4 inline" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round"><path stroke="none" d="M0 0h24v24H0z" fill="none"/><path d="M3 12a9 9 0 1 0 18 0a9 9 0 0 0 -18 0" /><path d="M9 12h6" /><path d="M12 9v6" /></svg>
-        </button>
+        </a>
     </div>
 
     <table class="min-w-full bg-gray-800 text-white">
