diff --git a/templates/voyage/new.html.twig b/templates/voyage/new.html.twig
index 79cfe75..9de7044 100644
--- a/templates/voyage/new.html.twig
+++ b/templates/voyage/new.html.twig
@@ -40,4 +40,8 @@
             {{ include('voyage/_row.html.twig') }}
         </template>
     </turbo-stream>
+
+    <turbo-stream action="update" targets="dialog[open] [data-modal-target=modalContent]">
+        <template></template>
+    </turbo-stream>
 {% endblock %}
