diff --git a/src/Twig/AppExtension.php b/src/Twig/AppExtension.php
new file mode 100644
index 0000000..5f4892d
--- /dev/null
+++ b/src/Twig/AppExtension.php
@@ -0,0 +1,35 @@
+<?php
+
+namespace App\Twig;
+
+use Symfony\Component\HttpFoundation\RequestStack;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+
+class AppExtension extends AbstractExtension
+{
+    public function __construct(private RequestStack $requestStack)
+    {
+    }
+
+    public function getFunctions()
+    {
+        return [
+            'is_frame_request' => new TwigFunction('is_frame_request', [$this, 'isFrameRequest']),
+        ];
+    }
+
+    public function isFrameRequest(?string $frameId = null): bool
+    {
+        $request = $this->requestStack->getCurrentRequest();
+        if (!$request->headers->has('Turbo-Frame')) {
+            return false;
+        }
+
+        if (null === $frameId) {
+            return true;
+        }
+
+        return $request->headers->get('Turbo-Frame') === $frameId;
+    }
+}
diff --git a/templates/voyage/new.html.twig b/templates/voyage/new.html.twig
index 9e7a391..fd201ae 100644
--- a/templates/voyage/new.html.twig
+++ b/templates/voyage/new.html.twig
@@ -3,7 +3,7 @@
 {% block title %}New Voyage{% endblock %}
 
 {% block body %}
-    <turbo-frame id="modal" target="_top" class="block m-4 p-4 bg-gray-800 rounded-lg">
+    <turbo-frame id="{{ is_frame_request('modal') ? 'modal' : 'modal-page' }}" target="_top" class="block m-4 p-4 bg-gray-800 rounded-lg">
         <div class="flex justify-between">
             <h1 class="text-xl font-semibold text-white mb-4">Create New Voyage</h1>
 
