diff --git a/templates/base.html.twig b/templates/base.html.twig
index b197a6d..bd803fc 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -52,5 +52,14 @@
         <div id="flash-container">
             {{ include('_flashes.html.twig') }}
         </div>
+        <twig:Modal>
+            <twig:block name="full_content">
+                <turbo-frame
+                    id="modal"
+                    data-modal-target="modalContent"
+                    data-action="turbo:before-fetch-request->modal#showLoading"
+                ></turbo-frame>
+            </twig:block>
+        </twig:Modal>
     </body>
 </html>
diff --git a/templates/components/Modal.html.twig b/templates/components/Modal.html.twig
index d09dc69..3dfce91 100644
--- a/templates/components/Modal.html.twig
+++ b/templates/components/Modal.html.twig
@@ -11,11 +11,11 @@
     >
         <div class="flex grow p-5">
             <div class="grow overflow-auto p-1">
-                <turbo-frame
-                    id="modal"
-                    data-modal-target="modalContent"
-                    data-action="turbo:before-fetch-request->modal#showLoading"
-                ></turbo-frame>
+                {% block full_content %}
+                    <div data-modal-target="modalContent">
+                        {% block content %}{% endblock %}
+                    </div>
+                {% endblock %}
             </div>
         </div>
     </dialog>
