diff --git a/src/Controller/VoyageController.php b/src/Controller/VoyageController.php
index f7d3fd4..e772dfd 100644
--- a/src/Controller/VoyageController.php
+++ b/src/Controller/VoyageController.php
@@ -11,6 +11,7 @@ use Symfony\Component\Form\FormInterface;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Annotation\Route;
+use Symfony\UX\Turbo\TurboBundle;
 
 #[Route('/voyage')]
 class VoyageController extends AbstractController
@@ -37,11 +38,11 @@ class VoyageController extends AbstractController
             $this->addFlash('success', 'Bon voyage!');
 
             if ($request->headers->has('turbo-frame')) {
-                $stream = $this->renderBlockView('voyage/new.html.twig', 'stream_success', [
+                $request->setRequestFormat(TurboBundle::STREAM_FORMAT);
+
+                return $this->renderBlock('voyage/new.html.twig', 'stream_success', [
                     'voyage' => $voyage
                 ]);
-
-                $this->addFlash('stream', $stream);
             }
 
             return $this->redirectToRoute('app_voyage_index', [], Response::HTTP_SEE_OTHER);
