diff --git a/assets/controllers/modal_controller.js b/assets/controllers/modal_controller.js
index d3784e8..37ef547 100644
--- a/assets/controllers/modal_controller.js
+++ b/assets/controllers/modal_controller.js
@@ -1,7 +1,7 @@
 import { Controller } from '@hotwired/stimulus';
 
 export default class extends Controller {
-    static targets = ['dialog', 'modalContent'];
+    static targets = ['dialog', 'modalContent', 'loadingTemplate'];
 
     observer = null;
 
@@ -42,4 +42,13 @@ export default class extends Controller {
             this.dialogTarget.close();
         }
     }
+
+    showLoading(event) {
+        // do nothing if the dialog is already open
+        if (this.dialogTarget.open) {
+            return;
+        }
+
+        this.modalContentTarget.innerHTML = this.loadingTemplateTarget.innerHTML;
+    }
 }
diff --git a/src/Controller/VoyageController.php b/src/Controller/VoyageController.php
index 47acff2..23b9290 100644
--- a/src/Controller/VoyageController.php
+++ b/src/Controller/VoyageController.php
@@ -31,6 +31,7 @@ class VoyageController extends AbstractController
         $voyage = new Voyage();
         $form = $this->createVoyageForm($voyage);
         $form->handleRequest($request);
+        sleep(2);
 
         if ($form->isSubmitted() && $form->isValid()) {
             $entityManager->persist($voyage);
diff --git a/templates/base.html.twig b/templates/base.html.twig
index 8276d36..86285d1 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -64,10 +64,30 @@
             >
                 <div class="flex grow p-5">
                     <div class="grow overflow-auto p-1">
-                        <turbo-frame id="modal" {{ stimulus_target('modal', 'modalContent') }}></turbo-frame>
+                        <turbo-frame
+                            id="modal"
+                            {{ stimulus_target('modal', 'modalContent') }}
+                            {{ stimulus_action('modal', 'showLoading', 'turbo:before-fetch-request') }}
+                        ></turbo-frame>
                     </div>
                 </div>
             </dialog>
+
+            <template {{ stimulus_target('modal', 'loadingTemplate') }}>
+                <div class="bg-space-pattern bg-cover rounded-lg p-8">
+                    <div class="space-y-2">
+                        <div class="h-4 bg-gray-700 rounded w-3/4 animate-pulse"></div>
+                        <div class="h-4 bg-gray-700 rounded animate-pulse"></div>
+                        <div class="h-4 bg-gray-700 rounded animate-pulse"></div>
+                        <div class="h-4"></div>
+                        <div class="h-4 bg-gray-700 rounded animate-pulse"></div>
+                        <div class="h-4 bg-gray-700 rounded w-1/2 animate-pulse"></div>
+                        <div class="h-4 bg-gray-700 rounded w-3/4 animate-pulse"></div>
+                        <div class="h-4"></div>
+                        <div class="h-4 bg-gray-700 rounded w-1/2 animate-pulse"></div>
+                    </div>
+                </div>
+            </template>
         </div>
     </body>
 </html>
