diff --git a/assets/app.js b/assets/app.js
index e3b03ade..4d2296d5 100644
--- a/assets/app.js
+++ b/assets/app.js
@@ -5,5 +5,6 @@
  * which should already be in your base.html.twig.
  */
 import './styles/app.css'
+import alienGreeting from './lib/alien-greeting.js';
 
-console.log('This log comes from assets/app.js - welcome to AssetMapper! 🎉')
+alienGreeting('Give us all your candy!', false);
diff --git a/templates/base.html.twig b/templates/base.html.twig
index 9947d500..554dae09 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -10,12 +10,6 @@
 
         {% block javascripts %}
             {{ importmap('app') }}
-
-            <script type="module">
-                import alienGreeting from '{{ asset('lib/alien-greeting.js') }}';
-
-                alienGreeting('Give us all your candy!', false);
-            </script>
         {% endblock %}
     </head>
     <body>
