diff --git a/templates/base.html.twig b/templates/base.html.twig
index 554dae09..9947d500 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -10,6 +10,12 @@
 
         {% block javascripts %}
             {{ importmap('app') }}
+
+            <script type="module">
+                import alienGreeting from '{{ asset('lib/alien-greeting.js') }}';
+
+                alienGreeting('Give us all your candy!', false);
+            </script>
         {% endblock %}
     </head>
     <body>
