diff --git a/assets/app.js b/assets/app.js
index 4d2296d..47b9f0f 100644
--- a/assets/app.js
+++ b/assets/app.js
@@ -4,7 +4,7 @@
  * This file will be included onto the page via the importmap() Twig function,
  * which should already be in your base.html.twig.
  */
-import './styles/app.css'
-import alienGreeting from './lib/alien-greeting.js';
+import './styles/app.css';
+import alienGreeting from './lib/alien-greeting';
 
 alienGreeting('Give us all your candy!', false);
