diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
index bb98d74..29fd8d9 100644
--- a/templates/main/homepage.html.twig
+++ b/templates/main/homepage.html.twig
@@ -57,7 +57,7 @@
                                             data-popover-target="card"
                                             class="max-w-sm rounded shadow-lg bg-gray-900 absolute left-0 bottom-8"
                                         >
-                                            <turbo-frame id="planet-card-{{ voyage.planet.id }}" target="_top" src="{{ path('app_planet_show_card', {
+                                            <turbo-frame loading="lazy" id="planet-card-{{ voyage.planet.id }}" target="_top" src="{{ path('app_planet_show_card', {
                                                 'id': voyage.planet.id,
                                             }) }}"></turbo-frame>
                                         </div>
