diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
index ffd03ae..ae2a7d8 100644
--- a/templates/main/homepage.html.twig
+++ b/templates/main/homepage.html.twig
@@ -59,7 +59,14 @@
                                         >
                                             <turbo-frame loading="lazy" id="planet-card-{{ voyage.planet.id }}" target="_top" src="{{ path('app_planet_show_card', {
                                                 'id': voyage.planet.id,
-                                            }) }}"></turbo-frame>
+                                            }) }}">
+                                                <div class="p-10">
+                                                    <svg xmlns="http://www.w3.org/2000/svg" class="animate-spin" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
+                                                       <path stroke="none" d="M0 0h24v24H0z" fill="none"></path>
+                                                       <path d="M12 3a9 9 0 1 0 9 9"></path>
+                                                    </svg>
+                                                </div>
+                                            </turbo-frame>
                                         </div>
                                     </template>
                                 </div>
