diff --git a/src/Controller/PlanetController.php b/src/Controller/PlanetController.php
index 1c5a760..e1f85f7 100644
--- a/src/Controller/PlanetController.php
+++ b/src/Controller/PlanetController.php
@@ -52,6 +52,14 @@ class PlanetController extends AbstractController
         ]);
     }
 
+    #[Route('/{id}/card', name: 'app_planet_show_card', methods: ['GET'])]
+    public function showCard(Planet $planet): Response
+    {
+        return $this->render('planet/_card.html.twig', [
+            'planet' => $planet,
+        ]);
+    }
+
     #[Route('/{id}/edit', name: 'app_planet_edit', methods: ['GET', 'POST'])]
     public function edit(Request $request, Planet $planet, EntityManagerInterface $entityManager): Response
     {
diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
index 40059a4..7659797 100644
--- a/templates/main/homepage.html.twig
+++ b/templates/main/homepage.html.twig
@@ -57,14 +57,9 @@
                                             data-popover-target="card"
                                             class="max-w-sm rounded shadow-lg bg-gray-900 absolute left-0 bottom-8"
                                         >
-                                            <div class="px-6 py-4">
-                                                <h4>
-                                                    <a class="hover:text-blue-300 transition-colors duration-100" href="{{ path('app_planet_show', { id: voyage.planet.id }) }}">
-                                                        {{ voyage.planet.name }}
-                                                    </a>
-                                                </h4>
-                                                <small>{{ voyage.planet.lightYearsFromEarth|round|number_format }} ly</small>
-                                            </div>
+                                            <turbo-frame id="planet-card-{{ voyage.planet.id }}" src="{{ path('app_planet_show_card', {
+                                                'id': voyage.planet.id,
+                                            }) }}"></turbo-frame>
                                         </div>
                                     </template>
                                 </div>
diff --git a/templates/planet/_card.html.twig b/templates/planet/_card.html.twig
new file mode 100644
index 0000000..0f2fecb
--- /dev/null
+++ b/templates/planet/_card.html.twig
@@ -0,0 +1,10 @@
+<turbo-frame id="planet-card-{{ planet.id }}">
+    <div class="px-6 py-4">
+        <h4>
+            <a class="hover:text-blue-300 transition-colors duration-100" href="{{ path('app_planet_show', { id: voyage.planet.id }) }}">
+                {{ voyage.planet.name }}
+            </a>
+        </h4>
+        <small>{{ voyage.planet.lightYearsFromEarth|round|number_format }} ly</small>
+    </div>
+</turbo-frame>
