diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
index c58b9bd7..40059a4e 100644
--- a/templates/main/homepage.html.twig
+++ b/templates/main/homepage.html.twig
@@ -55,7 +55,7 @@
                                     <template data-popover-target="content">
                                         <div
                                             data-popover-target="card"
-                                            class="bg-gray-900 absolute"
+                                            class="max-w-sm rounded shadow-lg bg-gray-900 absolute left-0 bottom-8"
                                         >
                                             <div class="px-6 py-4">
                                                 <h4>
