diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
index 1c0a68e..fdc5bc5 100644
--- a/templates/main/homepage.html.twig
+++ b/templates/main/homepage.html.twig
@@ -41,11 +41,36 @@
                         <tr class="border-b border-gray-700 {% if loop.index is odd %} bg-gray-800 {% else %} bg-gray-700 {% endif %}">
                             <td class="p-4">{{ voyage.purpose }}</td>
                             <td class="px-2 whitespace-nowrap">
-                                <img
-                                    src="{{ asset('images/'~voyage.planet.imageFilename) }}"
-                                    alt="Image of {{ voyage.planet.name }}"
-                                    class="inline-block w-8 h-8 rounded-full bg-gray-600 ml-2"
+                                <div
+                                    {{ stimulus_controller('popover') }}
+                                    {{
+                                        stimulus_action('popover', 'show', 'mouseenter')
+                                        |stimulus_action('popover', 'hide', 'mouseleave')
+                                    }}
+                                    class="relative"
                                 >
+                                    <img
+                                        src="{{ asset('images/'~voyage.planet.imageFilename) }}"
+                                        alt="Image of {{ voyage.planet.name }}"
+                                        class="inline-block w-8 h-8 rounded-full bg-gray-600 ml-2"
+                                    >
+
+                                    <template {{ stimulus_target('popover', 'content') }}>
+                                        <div
+                                            {{ stimulus_target('popover', 'card') }}
+                                            class="bg-gray-900 absolute"
+                                        >
+                                            <div class="px-6 py-4">
+                                                <h4>
+                                                    <a class="hover:text-blue-300 transition-colors duration-100" href="{{ path('app_planet_show', { id: voyage.planet.id }) }}">
+                                                        {{ voyage.planet.name }}
+                                                    </a>
+                                                </h4>
+                                                <small>{{ voyage.planet.lightYearsFromEarth|round|number_format }} ly</small>
+                                            </div>
+                                        </div>
+                                    </template>
+                                </div>
                             </td>
                             <td class="px-2 whitespace-nowrap">{{ voyage.leaveAt|ago }}</td>
                         </tr>
