diff --git a/assets/app.js b/assets/app.js
index 03295f9..4237e72 100644
--- a/assets/app.js
+++ b/assets/app.js
@@ -7,9 +7,5 @@ import './bootstrap.js';
  */
 import './styles/app.css';
 import alienGreeting from './lib/alien-greeting.js';
-import JSConfetti from 'js-confetti';
-
-const jsConfetti = new JSConfetti();
-jsConfetti.addConfetti();
 
 alienGreeting('Give us all your candy!', false);
diff --git a/assets/controllers/celebrate_controller.js b/assets/controllers/celebrate_controller.js
new file mode 100644
index 0000000..832f17b
--- /dev/null
+++ b/assets/controllers/celebrate_controller.js
@@ -0,0 +1,9 @@
+import { Controller } from '@hotwired/stimulus';
+import JSConfetti from 'js-confetti';
+
+export default class extends Controller {
+    poof() {
+        const jsConfetti = new JSConfetti();
+        jsConfetti.addConfetti();
+    }
+}
diff --git a/templates/base.html.twig b/templates/base.html.twig
index 272a8c0..b0e58b8 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -17,7 +17,10 @@
                 <nav class="flex items-center justify-between mb-4">
                     <div class="flex items-center">
                         <!-- Placeholder for the logo -->
-                        <svg class="h-8 w-8 text-green-500" fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 4v16m8-8H4"></path></svg>
+                        <svg
+                            {{ stimulus_controller('celebrate') }}
+                            {{ stimulus_action('celebrate', 'poof', 'mouseover') }}
+                            class="h-8 w-8 text-green-500" fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 4v16m8-8H4"></path></svg>
                         <a href="{{ path('app_homepage') }}" class="text-xl ml-3">Space Inviters</a>
                         <a href="{{ path('app_voyage_index') }}" class="ml-6 hover:text-gray-400">Voyages</a>
                         <a href="{{ path('app_planet_index') }}" class="ml-4 hover:text-gray-400">Planets</a>
