diff --git a/assets/app.js b/assets/app.js
index 03295f9..4237e72 100644
--- a/assets/app.js
+++ b/assets/app.js
@@ -7,9 +7,5 @@ import './bootstrap.js';
  */
 import './styles/app.css';
 import alienGreeting from './lib/alien-greeting.js';
-import JSConfetti from 'js-confetti';
-
-const jsConfetti = new JSConfetti();
-jsConfetti.addConfetti();
 
 alienGreeting('Give us all your candy!', false);
diff --git a/assets/controllers/celebrate_controller.js b/assets/controllers/celebrate_controller.js
new file mode 100644
index 0000000..832f17b
--- /dev/null
+++ b/assets/controllers/celebrate_controller.js
@@ -0,0 +1,9 @@
+import { Controller } from '@hotwired/stimulus';
+import JSConfetti from 'js-confetti';
+
+export default class extends Controller {
+    poof() {
+        const jsConfetti = new JSConfetti();
+        jsConfetti.addConfetti();
+    }
+}
diff --git a/templates/base.html.twig b/templates/base.html.twig
index 16ba943..6370838 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -17,7 +17,11 @@
             <header class="my-8 px-4">
                 <nav class="flex items-center justify-between mb-4">
                     <div class="flex items-center">
-                        <a href="{{ path    ('app_homepage') }}">
+                        <a
+                            href="{{ path    ('app_homepage') }}"
+                            {{ stimulus_controller('celebrate') }}
+                            {{ stimulus_action('celebrate', 'poof', 'mouseover') }}
+                        >
                             <img src="{{ asset('images/logo.png') }}" width="50" alt="Space Inviters Logo" >
                         </a>
                         <a href="{{ path('app_homepage') }}" class="text-xl ml-3">Space Inviters</a>
