diff --git a/templates/base.html.twig b/templates/base.html.twig
index 96d17132..26fa7096 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -6,7 +6,6 @@
         <title>{% block title %}Space Inviters!{% endblock %}</title>
         <link rel="icon" href="data:image/svg+xml,<svg xmlns=%22http://www.w3.org/2000/svg%22 viewBox=%220 0 128 128%22><text y=%221.2em%22 font-size=%2296%22>⚫️</text></svg>">
         {% block stylesheets %}
-            {{ ux_controller_link_tags() }}
         {% endblock %}
 
         {% block javascripts %}
