diff --git a/src/Controller/VoyageController.php b/src/Controller/VoyageController.php
index d7c8563..5b4b249 100644
--- a/src/Controller/VoyageController.php
+++ b/src/Controller/VoyageController.php
@@ -11,6 +11,7 @@ use Symfony\Component\Form\FormInterface;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Annotation\Route;
+use Symfony\UX\Turbo\TurboBundle;
 
 #[Route('/voyage')]
 class VoyageController extends AbstractController
@@ -37,6 +38,16 @@ class VoyageController extends AbstractController
 
             $this->addFlash('success', 'Bon voyage!');
 
+            if (TurboBundle::STREAM_FORMAT === $request->getPreferredFormat()) {
+                // If the request comes from Turbo, set the content type as text/vnd.turbo-stream.html and only send the HTML to update
+                $request->setRequestFormat(TurboBundle::STREAM_FORMAT);
+                $form = $this->createVoyageForm();
+
+                return $this->renderBlock('voyage/new.html.twig', 'success_stream', [
+                    'form' => $form
+                ]);
+            }
+
             return $this->redirectToRoute('app_voyage_show', ['id' => $voyage->getId()], Response::HTTP_SEE_OTHER);
         }
 
diff --git a/templates/voyage/new.html.twig b/templates/voyage/new.html.twig
index 95d416a..4cee8d0 100644
--- a/templates/voyage/new.html.twig
+++ b/templates/voyage/new.html.twig
@@ -21,3 +21,11 @@
         {{ include('voyage/_form.html.twig') }}
     </div>
 {% endblock %}
+
+{% block success_stream %}
+    <turbo-stream action="replace" target="voyage-form">
+        <template>
+            {{ include('voyage/_form.html.twig') }}
+        </template>
+    </turbo-stream>
+{% endblock %}
