diff --git a/src/Controller/VoyageController.php b/src/Controller/VoyageController.php
index c80b4d8..810fb19 100644
--- a/src/Controller/VoyageController.php
+++ b/src/Controller/VoyageController.php
@@ -18,7 +18,8 @@ class VoyageController extends AbstractController
     public function index(VoyageRepository $voyageRepository): Response
     {
         return $this->render('voyage/index.html.twig', [
-            'voyages' => $voyageRepository->findBy([], ['id' => 'DESC'])
+            'voyages' => $voyageRepository->findBy([], ['id' => 'DESC']),
+            'form' => $this->createForm(VoyageType::class),
         ]);
     }
 
diff --git a/templates/voyage/index.html.twig b/templates/voyage/index.html.twig
index 600d9c7..8beaed4 100644
--- a/templates/voyage/index.html.twig
+++ b/templates/voyage/index.html.twig
@@ -6,6 +6,8 @@
 <div class="m-4 p-4 bg-gray-800 rounded-lg">
     <h1 class="text-xl font-semibold text-white mb-4">Voyages</h1>
 
+    {{ include('voyage/_form.html.twig') }}
+
     <table class="min-w-full bg-gray-800 text-white">
         <thead class="border-b border-gray-700">
             <tr>
