diff --git a/src/Controller/VoyageController.php b/src/Controller/VoyageController.php
index af12546..b23cec3 100644
--- a/src/Controller/VoyageController.php
+++ b/src/Controller/VoyageController.php
@@ -78,6 +78,13 @@ class VoyageController extends AbstractController
 
             $this->addFlash('success', 'Voyage updated!');
 
+            if (TurboBundle::STREAM_FORMAT === $request->getPreferredFormat()) {
+                // If the request comes from Turbo, set the content type as text/vnd.turbo-stream.html and only send the HTML to update
+                $request->setRequestFormat(TurboBundle::STREAM_FORMAT);
+
+                return $this->renderBlock('voyage/edit.html.twig', 'success_stream');
+            }
+
             return $this->redirectToRoute('app_voyage_index', [], Response::HTTP_SEE_OTHER);
         }
 
diff --git a/templates/voyage/edit.html.twig b/templates/voyage/edit.html.twig
index 9c0a403..81e13f9 100644
--- a/templates/voyage/edit.html.twig
+++ b/templates/voyage/edit.html.twig
@@ -22,3 +22,11 @@
         {{ include('voyage/_form.html.twig') }}
     </div>
 {% endblock %}
+
+{% block success_stream %}
+    <turbo-stream action="append" target="flash-container">
+        <template>
+            {{ include('_flashes.html.twig') }}
+        </template>
+    </turbo-stream>
+{% endblock %}
