diff --git a/templates/voyage/new.html.twig b/templates/voyage/new.html.twig
index 4cee8d0..a4be319 100644
--- a/templates/voyage/new.html.twig
+++ b/templates/voyage/new.html.twig
@@ -28,4 +28,10 @@
             {{ include('voyage/_form.html.twig') }}
         </template>
     </turbo-stream>
+
+    <turbo-stream action="append" target="flash-container">
+        <template>
+            {{ include('_flashes.html.twig') }}
+        </template>
+    </turbo-stream>
 {% endblock %}
