diff --git a/tests/Functional/VoyageControllerTest.php b/tests/Functional/VoyageControllerTest.php
index 7bee4b07..5b35f7bf 100644
--- a/tests/Functional/VoyageControllerTest.php
+++ b/tests/Functional/VoyageControllerTest.php
@@ -23,7 +23,6 @@ class VoyageControllerTest extends AppPantherTestCase
         $this->pantherBrowser()
             ->visit('/')
             ->click('Voyages')
-            ->ddScreenshot()
             ->waitForPageLoad()
             ->click('New Voyage')
             ->fillField('Purpose', 'Test voyage')
