diff --git a/tests/Functional/VoyageControllerTest.php b/tests/Functional/VoyageControllerTest.php
new file mode 100644
index 0000000..74e82e5
--- /dev/null
+++ b/tests/Functional/VoyageControllerTest.php
@@ -0,0 +1,36 @@
+<?php
+
+namespace App\Tests\Functional;
+
+use App\Factory\PlanetFactory;
+use App\Factory\VoyageFactory;
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+use Zenstruck\Browser\Test\HasBrowser;
+use Zenstruck\Foundry\Test\Factories;
+use Zenstruck\Foundry\Test\ResetDatabase;
+
+class VoyageControllerTest extends WebTestCase
+{
+    use ResetDatabase;
+    use Factories;
+    use HasBrowser;
+
+    public function testCreateVoyage()
+    {
+        PlanetFactory::createOne([
+            'name' => 'Earth',
+        ]);
+        VoyageFactory::createOne();
+
+        $this->browser()
+            ->visit('/')
+            ->click('Voyages')
+            ->click('New Voyage')
+            ->fillField('Purpose', 'Test voyage')
+            ->selectFieldOption('Planet', 'Earth')
+            ->click('Save')
+            ->assertElementCount('table tbody tr', 2)
+            ->assertSee('Bon voyage')
+        ;
+    }
+}
