diff --git a/assets/controllers/closeable_controller.js b/assets/controllers/closeable_controller.js
index 4345d109..39b87d0f 100644
--- a/assets/controllers/closeable_controller.js
+++ b/assets/controllers/closeable_controller.js
@@ -1,6 +1,18 @@
 import { Controller } from '@hotwired/stimulus';
 
 export default class extends Controller {
+    static values = {
+        autoClose: Number,
+    };
+
+    connect() {
+        if (this.autoCloseValue) {
+            setTimeout(() => {
+                this.close();
+            }, this.autoCloseValue);
+        }
+    }
+
     close() {
         this.element.remove();
     }
diff --git a/templates/_flashes.html.twig b/templates/_flashes.html.twig
index 2881b47b..afa713b1 100644
--- a/templates/_flashes.html.twig
+++ b/templates/_flashes.html.twig
@@ -3,6 +3,7 @@
         class="fixed top-5 right-5 flex items-center w-full max-w-xs p-4 mb-4 text-gray-500 bg-white rounded-lg shadow dark:text-gray-400 dark:bg-gray-800"
         role="alert"
         data-controller="closeable"
+        data-closeable-auto-close-value="5000"
     >
         <div class="inline-flex items-center justify-center flex-shrink-0 w-8 h-8 text-green-500 bg-green-100 rounded-lg dark:bg-green-800 dark:text-green-200">
             <svg class="w-5 h-5" aria-hidden="true" xmlns="http://www.w3.org/2000/svg" fill="currentColor" viewBox="0 0 20 20">
