diff --git a/templates/_flashes.html.twig b/templates/_flashes.html.twig
new file mode 100644
index 00000000..ba82d5e4
--- /dev/null
+++ b/templates/_flashes.html.twig
@@ -0,0 +1,5 @@
+{% for message in app.flashes('success') %}
+    <div class="fixed bottom-0 right-0 m-4 p-4 bg-green-500 text-white rounded shadow">
+        {{ message }}
+    </div>
+{% endfor %}
diff --git a/templates/base.html.twig b/templates/base.html.twig
index e288055f..b197a6de 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -48,5 +48,9 @@
                 </div>
             </footer>
         </div>
+
+        <div id="flash-container">
+            {{ include('_flashes.html.twig') }}
+        </div>
     </body>
 </html>
