diff --git a/assets/controllers/closeable_controller.js b/assets/controllers/closeable_controller.js
index f3fba33..3a84766 100644
--- a/assets/controllers/closeable_controller.js
+++ b/assets/controllers/closeable_controller.js
@@ -5,11 +5,19 @@ export default class extends Controller {
         auto: Number,
     };
 
+    static targets = ['timerbar']
+
     connect() {
         if (this.autoValue) {
             setTimeout(() => {
                 this.close();
             }, this.autoValue);
+
+            if (this.hasTimerbarTarget) {
+                setTimeout(() => {
+                    this.timerbarTarget.style.width = 0;
+                }, 10);
+            }
         }
     }
 
diff --git a/templates/_flashes.html.twig b/templates/_flashes.html.twig
index f372dd9..ee61dbf 100644
--- a/templates/_flashes.html.twig
+++ b/templates/_flashes.html.twig
@@ -24,5 +24,10 @@
                 <path stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="m1 1 6 6m0 0 6 6M7 7l6-6M7 7l-6 6"/>
             </svg>
         </button>
+
+        <div
+            class="absolute bottom-0 left-0 h-1 bg-green-500 w-full transition-all duration-[5000ms] ease-linear"
+            data-closeable-target="timerbar"
+        ></div>
     </div>
 {% endfor %}
