diff --git a/assets/controllers/closeable_controller.js b/assets/controllers/closeable_controller.js
index 108e7b83..ecb60f20 100644
--- a/assets/controllers/closeable_controller.js
+++ b/assets/controllers/closeable_controller.js
@@ -30,6 +30,10 @@ export default class extends Controller {
     }
 
     close() {
+        if (this.hasTimerbarTarget) {
+            this.timerbarTarget.remove();
+        }
+
         this.leave();
     }
 }
