diff --git a/assets/controllers/closeable_controller.js b/assets/controllers/closeable_controller.js
index 975b77e..138f526 100644
--- a/assets/controllers/closeable_controller.js
+++ b/assets/controllers/closeable_controller.js
@@ -23,6 +23,9 @@ export default class extends Controller {
 
     close() {
         this.element.style.opacity = 0;
+        if (this.hasTimerbarTarget) {
+            this.timerbarTarget.remove();
+        }
 
         let handler;
         this.element.addEventListener('transitionend', handler = (event) => {
