diff --git a/assets/controllers/closeable_controller.js b/assets/controllers/closeable_controller.js
index 1ec887d..2dd2b00 100644
--- a/assets/controllers/closeable_controller.js
+++ b/assets/controllers/closeable_controller.js
@@ -1,4 +1,5 @@
 import { Controller } from '@hotwired/stimulus';
+import { useTransition } from 'stimulus-use';
 
 export default class extends Controller {
     static values = {
@@ -8,6 +9,13 @@ export default class extends Controller {
     static targets = ['timerbar']
 
     connect() {
+        useTransition(this, {
+            leaveActive: 'transition ease-in duration-200',
+            leaveFrom: 'opacity-100',
+            leaveTo: 'opacity-0',
+            transitioned: true,
+        });
+
         if (this.autoCloseValue) {
             setTimeout(() => {
                 this.close();
@@ -22,6 +30,6 @@ export default class extends Controller {
     }
 
     close() {
-        this.element.remove();
+        this.leave();
     }
 }
