diff --git a/templates/base.html.twig b/templates/base.html.twig
index e894f51..46c54fd 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -25,7 +25,7 @@
                             <img src="{{ asset('images/logo.png') }}" width="50" alt="Space Inviters Logo" >
                         </a>
                         <a href="{{ path('app_homepage') }}" class="text-xl ml-3">Space Inviters</a>
-                        <a href="{{ path('app_voyage_index') }}" class="ml-6 hover:text-gray-400">Voyages</a>
+                        <a href="{{ path('app_voyage_index') }}" data-turbo-preload class="ml-6 hover:text-gray-400">Voyages</a>
                         <a href="{{ path('app_planet_index') }}" class="ml-4 hover:text-gray-400">Planets</a>
                     </div>
                     <div
