diff --git a/src/Twig/Components/Button.php b/src/Twig/Components/Button.php
index f09ea9a..e239ebf 100644
--- a/src/Twig/Components/Button.php
+++ b/src/Twig/Components/Button.php
@@ -7,5 +7,5 @@ use Symfony\UX\TwigComponent\Attribute\AsTwigComponent;
 #[AsTwigComponent]
 class Button
 {
-
+    public string $type = 'success';
 }
diff --git a/templates/components/Button.html.twig b/templates/components/Button.html.twig
index 0053b91..9c30e97 100644
--- a/templates/components/Button.html.twig
+++ b/templates/components/Button.html.twig
@@ -1,3 +1,3 @@
 <button {{ attributes.defaults({
     class: 'inline-flex items-center px-4 py-2 border border-transparent text-sm font-medium rounded-md text-white bg-green-600 hover:bg-green-700',
-}) }}>{% block content %}{% endblock %}</button>
+}) }}>{{ type }}{% block content %}{% endblock %}</button>
diff --git a/templates/voyage/_delete_form.html.twig b/templates/voyage/_delete_form.html.twig
index 9154ba1..4160998 100644
--- a/templates/voyage/_delete_form.html.twig
+++ b/templates/voyage/_delete_form.html.twig
@@ -1,7 +1,7 @@
 <form method="post" action="{{ path('app_voyage_delete', {'id': voyage.id}) }}" onsubmit="return confirm('Are you sure you want to delete this item?');">
     <input type="hidden" name="_token" value="{{ csrf_token('delete' ~ voyage.id) }}">
 
-    <twig:Button class="text-white bg-red-600 hover:bg-red-700 focus:ring-4 focus:ring-red-300 focus:outline-none">
+    <twig:Button type="danger" class="text-white bg-red-600 hover:bg-red-700 focus:ring-4 focus:ring-red-300 focus:outline-none">
         Delete
     </twig:Button>
 </form>
