diff --git a/templates/voyage/_form.html.twig b/templates/voyage/_form.html.twig
index 1ea91ebb..dd5bed38 100644
--- a/templates/voyage/_form.html.twig
+++ b/templates/voyage/_form.html.twig
@@ -1,10 +1,10 @@
 {{ form_start(form) }}
     {{ form_widget(form) }}
 
-    {{ component('Button', {
-        formnovalidate: true,
-        class: 'px-4 py-2 border border-transparent text-sm font-medium rounded-md text-white bg-green-600 hover:bg-green-700',
-    }) }}
+    <twig:Button
+        formnovalidate
+        class="px-4 py-2 border border-transparent text-sm font-medium rounded-md text-white bg-green-600 hover:bg-green-700"
+    />
 
     <button formnovalidate class="px-4 py-2 border border-transparent text-sm font-medium rounded-md text-white bg-green-600 hover:bg-green-700">
         {{ button_label|default('Save') }}
