diff --git a/src/Twig/Components/Button.php b/src/Twig/Components/Button.php
new file mode 100644
index 00000000..f09ea9a5
--- /dev/null
+++ b/src/Twig/Components/Button.php
@@ -0,0 +1,11 @@
+<?php
+
+namespace App\Twig\Components;
+
+use Symfony\UX\TwigComponent\Attribute\AsTwigComponent;
+
+#[AsTwigComponent]
+class Button
+{
+
+}
diff --git a/templates/components/Button.html.twig b/templates/components/Button.html.twig
new file mode 100644
index 00000000..82eb1f31
--- /dev/null
+++ b/templates/components/Button.html.twig
@@ -0,0 +1 @@
+<button {{ attributes }}>Press me!</button>
diff --git a/templates/voyage/_form.html.twig b/templates/voyage/_form.html.twig
index 1c079b4a..d0e5351e 100644
--- a/templates/voyage/_form.html.twig
+++ b/templates/voyage/_form.html.twig
@@ -2,6 +2,11 @@
     {{ form_start(form) }}
         {{ form_widget(form) }}
 
+        {{ component('Button', {
+            formnovalidate: true,
+            class: 'px-4 py-2 border border-transparent text-sm font-medium rounded-md text-white bg-green-600 hover:bg-green-700',
+        }) }}
+
         <button formnovalidate class="px-4 py-2 border border-transparent text-sm font-medium rounded-md text-white bg-green-600 hover:bg-green-700">
             {{ button_label|default('Save') }}
         </button>
