diff --git a/templates/voyage/_form.html.twig b/templates/voyage/_form.html.twig
index 83ff5763..9fdbc880 100644
--- a/templates/voyage/_form.html.twig
+++ b/templates/voyage/_form.html.twig
@@ -7,8 +7,4 @@
     >
         {{ button_label|default('Save') }}
     </twig:Button>
-
-    <button formnovalidate class="px-4 py-2 border border-transparent text-sm font-medium rounded-md text-white bg-green-600 hover:bg-green-700">
-        {{ button_label|default('Save') }}
-    </button>
 {{ form_end(form) }}
