diff --git a/assets/app.js b/assets/app.js
index 65959540..3f6c690e 100644
--- a/assets/app.js
+++ b/assets/app.js
@@ -7,8 +7,9 @@ import { shouldPerformTransition, performTransition } from 'turbo-view-transitio
 //Turbo.session.drive = false;
 alienGreeting('Give us all your candy!', false);
 
+let skipNextRenderTransition = false;
 document.addEventListener('turbo:before-render', (event) => {
-    if (shouldPerformTransition()) {
+    if (shouldPerformTransition() && !skipNextRenderTransition) {
         event.preventDefault();
 
         performTransition(document.body, event.detail.newBody, async () => {
@@ -26,6 +27,13 @@ document.addEventListener('turbo:before-frame-render', (event) => {
     if (shouldPerformTransition()) {
         event.preventDefault();
 
+        // workaround for data-turbo-action="advance", which triggers
+        // turbo:before-render (and we want THAT to not try to transition)
+        skipNextRenderTransition = true;
+        setTimeout(() => {
+            skipNextRenderTransition = false;
+        }, 100);
+
         performTransition(event.target, event.detail.newFrame, async () => {
             await event.detail.resume();
         });
